assets/js/payment-method-extensions/payment-methods/stripe/payment-request/use-initialization.js

```diff
@@ -56,41 +53,30 @@ export const useInitialization = ( {
 
 	// Update refs when any change.
 	useEffect( () => {
-		currentPaymentRequest.current = paymentRequest;
 		currentShipping.current = shippingData;
-	}, [ paymentRequest, shippingData ] );
+	}, [ shippingData ] );
 
-	// set paymentRequest.
+	// Create the initial paymentRequest object. Note, we can't do anything if stripe isn't available yet or we have zero total.
 	useEffect( () => {
-		// can't do anything if stripe isn't available yet or we have zero total.
 		if ( ! stripe || ! billing.cartTotal.value ) {
 			return;
 		}
-
-		// if payment request hasn't been set yet then set it.
-		if ( ! currentPaymentRequest.current && ! isFinished ) {
-			setPaymentRequest(
-				getPaymentRequest( {
-					total: billing.cartTotal,
-					currencyCode: billing.currency.code.toLowerCase(),
-					countryCode: getSetting( 'baseLocation', {} )?.country,
-					shippingRequired: shippingData.needsShipping,
-					cartTotalItems: billing.cartTotalItems,
-					stripe,
-				} )
-			);
-		}
-		// otherwise we just update it (but only if payment processing hasn't
-		// already started).
-		if ( ! isProcessing && currentPaymentRequest.current && ! isFinished ) {
-			updatePaymentRequest( {
-				// @ts-ignore
-				paymentRequest: currentPaymentRequest.current,
-				total: billing.cartTotal,
-				currencyCode: billing.currency.code.toLowerCase(),
-				cartTotalItems: billing.cartTotalItems,
-			} );
+		if ( isFinished || isProcessing || paymentRequest ) {
+			return;
 		}
+		const pr = getPaymentRequest( {
```

Comment:
This new approach of recreating the request object seems much cleaner and simpler IMO, nice idea 👍